src/providers/fakerest/dataProvider.ts

```diff
@@ -241,6 +242,16 @@ export const dataProvider = withLifecycleCallbacks(
 
                 return params;
             },
+            afterSave: async data => {
+                const currentUser = await authProvider.getIdentity?.();
+                if (currentUser?.id === data.id) {
+                    localStorage.setItem(
+                        USER_STORAGE_KEY,
+                        JSON.stringify(data)
+                    );
+                }
+                return data;
+            },
```

Comment:
suggestion: add a little comment to explain why this is necessary? (it certainly is not obvious for me at least)